### ISSUE TITLE
warn if environment variable declarations are incorrectly quoted

### ISSUE THREAD

**Issue by HirenHiranandani-at** [koalaman/shellcheck#2491]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

"FOO=$(bar)" echo "$FOO"
```

#### Here's what shellcheck currently says:
```
No issues detected!
```

#### Here's what I wanted or expected to see:
```
Line 3:
"FOO=$(bar)" echo $FOO
^-- SCXXXX (info): quoting the variable name and value will not result in environment change 

Did you mean:
FOO="$(bar)" echo "$FOO"
```

**Comment by kurahaupo** [comment]
Suggest changing title of this issue:

> warn if environment prefix assignment is mistakenly quoted

1. there are no declarations in Shell, only assignments, attribute modifications, and unsetting.
1. “quoted incorrectly” implies that there's a way to quote it _correctly_, but the name and the `=` sign must be _unquoted_. “mistakenly quoted” clarifies that the use of quotes _was_ the error.

Contrast this with assignments that follow the special utility names `local`, `export`, `readonly`, `declare`.

**Comment by brother** [comment]
Executing the code as proposed would render errors, is it possible to construct a minimal example of a broken use that renders anything else than syntax errors?

**Comment by kurahaupo** [comment]
@brother It never reports a syntax error, merely "command not found"; so it's fairly trivial to arrange for the command _to_ be found:

Set up:

    foo=foo=bar   # repeating `foo=` is the initial error
    echo '#!/bin/echo' >$foo
    chmod a+x "$foo"
    PATH+=:

Works:

    foo=bar env | grep foo=

Misbehaves without failing:

    "foo=bar" env | grep foo=
